librz/arch/isa/msp430/msp430_il.c

```diff
@@ -0,0 +1,521 @@
+		SETL("source", get_source(op, current_addr)),
+		SETL("result", LOGXOR(VARL("source"), VARL("old_destination"))),
+
+		set_destination(op, VARL("result"), current_addr),
+		set_xor_flags("source", "old_destination", "result"));
+}
+
+RZ_OWN RzILOpEffect *rz_msp430_lift_mov_instr(RZ_BORROW RZ_NONNULL RzAnalysis *analysis, const Msp430Instruction *op, ut64 current_addr, int instr_size) {
+	return set_destination(op, get_source(op, current_addr), current_addr);
+}
+
+RZ_OWN RzILOpEffect *rz_msp430_lift_sub_or_cmp_instr(RZ_BORROW RZ_NONNULL RzAnalysis *analysis, const Msp430Instruction *op, bool write_dst, ut64 current_addr, int instr_size) {
+	RzILOpBitVector *increment_by;
+	RzILOpBitVector *carry_const;
+	RzILOpBitVector *overflow_const;
+	if (op->is_byte) {
+		increment_by = U8(1);
+		carry_const = U8(0);
+		overflow_const = U8(~0x7F);
+	} else {
+		increment_by = U16(1);
+		carry_const = U16(0);
+		overflow_const = U16(~0x7FFF);
+	}
+
+	RzILOpEffect *get_src = SETL("op0", get_source(op, current_addr));
+	RzILOpEffect *neg_src_add1 = SETL("op1", ADD(increment_by, LOGNOT(VARL("op0"))));
+	RzILOpEffect *get_dst = SETL("op2", get_destination(op, current_addr));
+	RzILOpEffect *compute_result = SETL("result", ADD(VARL("op1"), VARL("op2")));
+	RzILOpEffect *set_flags = set_sub_flags("op0", "op1", "op2", "result", carry_const, overflow_const);
+
+	return (write_dst) ? SEQ6( // sub
+				     get_src,
+				     neg_src_add1,
+				     get_dst,
+				     compute_result,
+				     set_destination(op, VARL("result"), current_addr),
+				     set_flags)
+			   : SEQ5( // cmp
+				     get_src,
+				     neg_src_add1,
+				     get_dst,
+				     compute_result,
+				     set_flags);
+}
+
+RZ_OWN RzILOpEffect *rz_msp430_lift_sub_instr(RZ_BORROW RZ_NONNULL RzAnalysis *analysis, const Msp430Instruction *op, ut64 current_addr, int instr_size) {
+	return rz_msp430_lift_sub_or_cmp_instr(analysis, op, true, current_addr, instr_size);
+}
+
+RZ_OWN RzILOpEffect *rz_msp430_lift_cmp_instr(RZ_BORROW RZ_NONNULL RzAnalysis *analysis, const Msp430Instruction *op, ut64 current_addr, int instr_size) {
+	return rz_msp430_lift_sub_or_cmp_instr(analysis, op, false, current_addr, instr_size);
+}
+
+RZ_OWN RzILOpEffect *rz_msp430_lift_bis_instr(RZ_BORROW RZ_NONNULL RzAnalysis *analysis, const Msp430Instruction *op, ut64 current_addr, int instr_size) {
+	RzILOpBitVector *result = LOGOR(get_source(op, current_addr), get_destination(op, current_addr));
+	return set_destination(op, result, current_addr);
+}
+
+RZ_OWN RzILOpEffect *rz_msp430_lift_bic_instr(RZ_BORROW RZ_NONNULL RzAnalysis *analysis, const Msp430Instruction *op, ut64 current_addr, int instr_size) {
+	RzILOpBitVector *result = LOGAND(LOGNOT(get_source(op, current_addr)), get_destination(op, current_addr));
+	return set_destination(op, result, current_addr);
+}
+/* @} */
+
+// ************************************* ---------------------------- ******************************** //
```

Comment:
You forgot to remove non-Doxygen comment lines then

Reply:
Can we leave those comment markers ? They're a very obvious way to demarcate the file into the conceptually seperate "Regions". Doxygen groups don't look as visually distinct. 

With those comments, I can look at the file overview in the small window in VSCode and immeditately know that the file is seperated into independent chunks.

Reply:
I don't strongly oppose it but it will be the only file that uses such a way to group, so it will be inconsistent with other code.

Reply:
Agree partially with @XVilka here. But I like these comment blocks as well. But one line is enough, IMHO.

Maybe without those ones?
```
// ************************************* ---------------------------- ******************************** //
```